src/conversion.rs

```diff
@@ -0,0 +1,46 @@
+use bitcoin::Network;
+
+/// Converter purpose is to give a Conversion from a given amount in satoshis according to its
+/// parameters.
+pub struct Converter {
+    pub unit: Unit,
+}
+
+impl Converter {
+    pub fn new(bitcoin_network: Network) -> Self {
+        let unit = match bitcoin_network {
+            Network::Testnet => Unit::TestnetBitcoin,
+            Network::Bitcoin => Unit::Bitcoin,
+            Network::Regtest => Unit::RegtestBitcoin,
+        };
+        Self { unit }
+    }
+
+    /// converts amount in satoshis to BTC float.
+    pub fn converts(&self, amount: &u64) -> f64 {
```

Comment:
What's the point of taking a reference to an `u64` ?  Even more if you copy it on the next line..